analysis/venn/venn.R

```diff
@@ -0,0 +1,258 @@
+)
+
+# Populate Venn table for each outcome -----------------------------------------
+print('Populate Venn table for each outcome')
+
+for (outcome in outcomes) {
+  print(paste0("Outcome: ", outcome))
+
+  # Load model input data ------------------------------------------------------
+  print('Load model input data')
+
+  model_input <- readr::read_rds(paste0(
+    "output/model/model_input-cohort_",
+    cohort,
+    "-",
+    analyses,
+    "-",
+    outcome,
+    ".rds"
+  ))
+  model_input <- model_input[
+    !is.na(model_input$out_date),
+    c("patient_id", "out_date")
+  ]
+
+  # Filter Venn data based on model input --------------------------------------
+  print('Filter Venn data based on model input')
+
+  tmp <- venn[
+    venn$patient_id %in% model_input$patient_id,
+    c("patient_id", colnames(venn)[grepl(outcome, colnames(venn))])
+  ]
+
+  colnames(tmp) <- gsub(
+    paste0("tmp_out_date_", outcome, "_"),
+    "",
+    colnames(tmp)
+  )
+
+  # Identify and add missing columns -------------------------------------------
+  print('Identify and add missing columns')
+
+  complete <- data.frame(
+    patient_id = tmp$patient_id,
+    gp = as.Date(NA),
+    apc = as.Date(NA),
+    death = as.Date(NA)
+  )
+
+  complete[, setdiff(colnames(tmp), "patient_id")] <- NULL
+  notused <- NULL
+
+  if (ncol(complete) > 1) {
+    tmp <- merge(tmp, complete, by = c("patient_id"))
+    notused <- setdiff(colnames(complete), "patient_id")
+  }
+
+  # Calculate the number contributing to each source combination ---------------
+  print('Calculate the number contributing to each source combination')
+
+  tmp$gp_contributing <- !is.na(tmp$gp) &
+    is.na(tmp$apc) &
+    is.na(tmp$death)
+
+  tmp$apc_contributing <- is.na(tmp$gp) &
+    !is.na(tmp$apc) &
+    is.na(tmp$death)
+
+  tmp$death_contributing <- is.na(tmp$gp) &
+    is.na(tmp$apc) &
+    !is.na(tmp$death)
+
+  tmp$gp_apc_contributing <- !is.na(tmp$gp) &
+    !is.na(tmp$apc) &
+    is.na(tmp$death)
+
+  tmp$apc_death_contributing <- is.na(tmp$gp) &
+    !is.na(tmp$apc) &
+    !is.na(tmp$death)
+
+  tmp$gp_death_contributing <- !is.na(tmp$gp) &
+    is.na(tmp$apc) &
+    !is.na(tmp$death)
+
+  tmp$gp_apc_death_contributing <- !is.na(tmp$gp) &
+    !is.na(tmp$apc) &
+    !is.na(tmp$death)
+
+  # Record the number contributing to each source combination ------------------
+  print('Record the number contributing to each source combination')
+
+  df[nrow(df) + 1, ] <- c(
+    outcome,
+    only_gp = nrow(tmp %>% filter(gp_contributing == T)),
+    only_apc = nrow(tmp %>% filter(apc_contributing == T)),
+    only_death = nrow(tmp %>% filter(death_contributing == T)),
+    gp_apc = nrow(tmp %>% filter(gp_apc_contributing == T)),
+    gp_death = nrow(tmp %>% filter(gp_death_contributing == T)),
+    apc_death = nrow(tmp %>% filter(apc_death_contributing == T)),
+    gp_apc_death = nrow(tmp %>% filter(gp_apc_death_contributing == T)),
+    total_gp = nrow(tmp %>% filter(!is.na(gp))),
```

Comment:
Remove totals - we could derive them but I don't think we need them (as they can be calculated outside if we want them) and it is fewer columns for output checking.